src/libp2p/Libp2p.Protocols.Ping/PingProtocol.cs

```diff
@@ -24,33 +27,48 @@ public PingProtocol(ILoggerFactory? loggerFactory = null)
     public async Task DialAsync(IChannel channel, IChannelFactory? channelFactory,
         IPeerContext context)
     {
-        byte[] bytes = new byte[32];
-        _random.NextBytes(bytes.AsSpan(0, 32));
-        _logger?.LogDebug("Ping {remotePeer}", context.RemotePeer.Address);
-        await channel.WriteAsync(new ReadOnlySequence<byte>(bytes));
-        _logger?.LogTrace("Reading pong");
-        ReadOnlySequence<byte> response = await channel.ReadAsync(32, ReadBlockingMode.WaitAll);
-        _logger?.LogTrace("Verifing pong");
-        if (!Enumerable.SequenceEqual(bytes[0..32], response.ToArray()))
+        byte[] byteArray = new byte[PayloadLength];
+        _random.NextBytes(byteArray.AsSpan(0, PayloadLength));
+        ReadOnlySequence<byte> bytes = new(byteArray);
+
+        _logger?.LogPing(context.RemotePeer.Address);
+        await channel.WriteAsync(bytes);
+
+        _logger?.ReadingPong();
+        ReadOnlySequence<byte> response = await channel.ReadAsync(PayloadLength, ReadBlockingMode.WaitAll);
+
+        _logger?.VerifyingPong();
+        if (!byteArray[0..PayloadLength].SequenceEqual(response.ToArray()))
         {
-            _logger?.LogWarning("Wrong response to ping from {from}", context.RemotePeer);
-            throw new ApplicationException();
+            // TODO: do we need log context.RemotePeer or context.RemotePeer.Address?
+            // _logger?.LogWarning("Wrong response to ping from {from}", context.RemotePeer);
+
+            // TODO: if we throw exception, probably LogLevel bigger than warning?
+            _logger?.PingFailed(context.RemotePeer.Address);
+            throw new ApplicationException(); // TODO: add specific exception?
         }
-        _logger?.LogDebug("Pinged");
+
+        _logger?.LogPinged(); // TODO: add context.RemotePeer.Address?
```

Comment:
Log above reveals it already, but let's add, which will simplify parallel pings debugging